lib/init/grass.py

```diff
@@ -264,6 +265,43 @@ def wxpath(*args):
     return os.path.join(_WXPYTHON_BASE, *args)
 
 
+def count_wide_chars(s):
+    """Returns the number of wide CJK characters in a string.
+
+    :param str s: string
+    """
+    return sum(unicodedata.east_asian_width(c) in 'WF' for c in
+            (s if sys.version_info.major >= 3 else unicode(s)))
+
+
+def f(fmt, *args):
+    """Adjusts fixed-width string specifiers for wide CJK characters and
+    returns a formatted string. Does not support named arguments yet.
+
+    :param str fmt: format string
+    :param *args: arguments for the format string
+    """
+    matches = []
+    # https://docs.python.org/3/library/stdtypes.html#old-string-formatting
+    for m in re.finditer('%([#0 +-]*)([0-9]*)(\.[0-9]*)?([hlL]?[diouxXeEfFgGcrsa%])', fmt):
+        matches.append(m)
+
+    if len(matches) != len(args):
+        raise Exception('The numbers of format specifiers and arguments do not match')
```

Comment:
```suggestion
        raise Exception('The length of format specifiers and arguments do not match')
```
(no idea, just to be more specific?)

Reply:
It's more of counts than lengths.

```python
print(f('Hello %s %d', 'world'))
```
The number of specifiers (`%s` and `%d`) is 2 while the number of arguments (`'world'`) is 1. They don't match. I think `number` is more clear (clearer?) than `length` in this case.

Now, my English is under test. "The number of format specifiers" and "the number of arguments" "do not match" => "The number**s** of format specifiers and arguments do not match" instead of "The number of format specifiers and arguments does not (or do not) match".